File: other/01/puzzle01.py
input01 = open('input01', encoding='utf-8').read().strip()

measurements = [int(line) for line in input01.split("\n")]

increases = 0
last_depth = measurements[0]
for depth in measurements[1:]:
    if depth > last_depth:
        increases += 1
    last_depth = depth

print(f"(p1 answer) increases = {increases}") #answer: 1548


increases = 0
last_window = measurements[0:3]
for i in range(1, len(measurements)-2):
    window = measurements[i:i+3]
    if sum(window) > sum(last_window):
        increases += 1
    last_window = window

print(f"(p2 answer) increases = {increases}") #answer: 1589

File: aoc2021kotlin/src/main/kotlin/Puzzle01.kt
package net.blergh.advent2021

import java.io.File

object Puzzle01 {
    fun part1(measurements: List<Int>) {
        var increases = 0
        var lastDepth = measurements.first()

        for (depth in measurements.drop(1)) {
            if (depth > lastDepth) {
                increases++
            }
            lastDepth = depth
        }

        println("(p1 answer) increases = $increases") //answer: 1548
    }

    fun part2(measurements: List<Int>) {
        var increases = 0
        var lastWindow = measurements.take(3)

        for (i in 1..(measurements.size - 3)) {
            val window = measurements.slice(i..i + 2)
            if (window.sum() > lastWindow.sum()) {
                increases++
            }
            lastWindow = window
        }

        println("(p2 answer) increases = $increases") //answer: 1589
    }

    fun run() {
        val input01 = File("${Main.aocRoot}/other/01/input01").readText().trim()
        val measurements = input01.split("\n").map(Integer::parseInt)

        part1(measurements)
        part2(measurements)
    }
}